test/Microsoft.ML.TestFramework/DataPipe/TestDataPipeBase.cs

```diff
@@ -1158,12 +1158,12 @@ protected Func<bool> GetColumnComparer(IRow r1, IRow r2, int col, ColumnType typ
             throw Contracts.Except("Unknown type in GetColumnComparer: '{0}'", type);
         }
 
-        private const Double DoubleEps = 1e-9;
+        private const Double DoubleEps = 1e-5;
 
         private static bool EqualWithEpsDouble(Double x, Double y)
         {
             // bitwise comparison is needed because Abs(Inf-Inf) and Abs(NaN-NaN) are not 0s.
-            return FloatUtils.GetBits(x) == FloatUtils.GetBits(y) || Math.Abs(x - y) < DoubleEps;
+            return FloatUtils.GetBits(x) == FloatUtils.GetBits(y) || Math.Abs(x - y) / Math.Abs(x) < DoubleEps;
```

Comment:
Why are you dividing by `abs(x)`?

Reply:
when the numbers are large like in hundreds or thousands,  just checking for an absolute error is causing unnecessary failures. Divinding by ```abs(x)``` gives us the measure of relative error

Reply:
We should probably be updating this to call into the comparison algorithm that was already written, which tries to account for these differences (and also gives a centralized place to update if it needs to be tweaked again in the future).

Reply:
are you talking about https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.TestFramework/BaseTestBaseline.cs#L547 comparison algo ?
I dont think that this one also doesn't handle the magnitude part

Reply:
The `CompareNumbersWithTolerance` function factors in both the `integer` and `fractional` part when doing the rounding and comparison.